ethash/src/compute.rs

```diff
@@ -85,6 +86,7 @@ pub type H256 = [u8; 32];
 pub struct Light {
 	block_number: u64,
 	cache: Vec<Node>,
+	eth_compute: Mutex<EthCompute>,
```

Comment:
This structure is not a member of `Engine`. It is managed by `EthashManager` only. Adding Mutex to eth.rs is fine because `Sync` is required there indeed. `Light` does not to be `Sync` though